### DISCUSSION TITLE
Plugin Loading

### DISCUSSION THREAD

**Discussion by AhmedThahir** [hakimel/reveal.js#3098] [Q&A] [answered]
1. Is there a way to load plugins conditionally, now that the dependencies loader method has been deprecated.
2. Is there a way to lazyload plugins like highlight.js?
    Deferring them causes only a blank white screen to appear.

**Comment by hakimel** [comment, marked as answer]
Yes, you can lazy or conditionally load plugins but you will need to call `Reveal.registerPlugin` to register them.

Here's an example:
```js
Reveal.initialize();

const loadScript = src => {
  return new Promise((resolve, reject) => {
      const script = document.createElement('script');
      script.type = 'text/javascript';
      script.onload = resolve;
      script.onerror = reject;
      script.src = src;
      document.head.append(script);
  })
}

loadScript( 'plugin/highlight/highlight.js' ).then(() => {
  Reveal.registerPlugin( RevealHighlight )
})
```
